### PR TITLE
PureScript 0.15 updates

### PR DESCRIPTION
**Pull request by ptrfrncsmrph**

### CODE REVIEW

**Comment by mjrussell** [comment]
Awesome, thanks @ptrfrncsmrph! I'll fix the travis/github ci stuff real quick before we get this in to make sure CI is passing

**Comment by mjrussell** [comment]
Actually since this adds spago (thanks for that!) maybe you want to do that as part of this PR. Here's an example, just remove the examples building at the bottom - https://github.com/lumihq/purescript-react-basic-classic/blob/main/.github/workflows/ci.yml

File: packages.dhall [hunk 1]
```diff
@@ -0,0 +1,138 @@
+{-
```
Comment (mjrussell) [inline packages.dhall:1]: Would you mind removing this and the comment from the spago.dhall also? They are a lot of noise that don't add that much value IMO

**Comment by ptrfrncsmrph** [comment]
Hmm, now I'm seeing a failure with
```
* ERROR: Your node.js version is too old (required: 12.0.0, actual: 10.24.1)
```
but I don't know where `10.24.1` would be specified

**Comment by ptrfrncsmrph** [comment]
Ah, I see now, it was Travis CI that's configured with old Node version
```json
{
  "language": "node_js",
  "os": [
    "linux"
  ],
  "dist": "trusty",
  "sudo": true,
  "node_js": [
    "10"
  ],
  "install": [
    "npm install"
  ],
  "script": [
    "npx bower install",
    "npm -s test"
  ]
}
```
I'm guessing Travis would have to be disabled by maintainer, and then the GitHub workflow could be approved and run?

**Comment by mjrussell** [comment]
> I'm guessing Travis would have to be disabled by maintainer, and then the GitHub workflow could be approved and run? (Sorry for noise of extraneous commits!)

No worries! If you want you can remove Travis in this PR (delete the .travis.yml file) or I can push that to main

File: examples/basic/README.md [hunk 2]
```diff
@@ -2,11 +2,13 @@
...
+You can build this example from the root of the `purescript-react-dnd-basic` project:
+
 ```sh
 npm install
-make all
+npm run example:basic
 ```
 
-This will compile the PureScript source files, bundle them, and use Browserify to combine PureScript and NPM sources into a single bundle.
+This will compile the PureScript source files, bundle them, and combine PureScript and NPM sources into a single bundle.
```
Comment (ptrfrncsmrph) [inline examples/basic/README.md:12]: Changed this to be a bit more like the Halogen examples: https://github.com/purescript-halogen/purescript-halogen/tree/master/examples/basic (removed the Makefile and `package.json` from this directory in favor of running `spago` from root directory and adding NPM deps to top-level `devDependencies`)

File: examples/basic/spago.dhall [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+              config.dependencies
+            # [ "arrays"
+              , "exceptions"
+              , "foldable-traversable"
+              , "integers"
+              , "react-basic"
+              , "react-basic-dom"
+              , "web-html"
+              ]
+        }
```
Comment (ptrfrncsmrph) [inline examples/basic/spago.dhall:15]: This formatting looks a bit strange, but this is how `dhall format` likes things 🤷

File: examples/basic/src/Basic.purs [hunk 4]
```diff
@@ -119,3 +125,15 @@ moveItem fromIndex toIndex items =
...
+main = do
+  maybeContainer <- window
+    >>= document
+    >>= toNonElementParentNode
+      >>> getElementById "container"
+  case maybeContainer of
+    Nothing -> throw "Container element not found."
+    Just container -> do
+      todoExample <- mkTodoExample
+      render (todoExample unit) container
```
Comment (ptrfrncsmrph) [inline examples/basic/src/Basic.purs:139]: This used to be in a separate `index.js`, and there was a Browserify command in the Makefile that would bundle it with the `purs` output (after writing `module.exports = PS.Basic;`). It seemed simpler to just use `spago bundle-app` with a single PS module.

**Comment by mjrussell** [comment]
Thanks for this!

**Pull request merged**